bundles/org.openhab.automation.scriptenginefactory.jython/src/main/java/org/openhab/automation/scriptenginefactory/jython/JythonScriptEngineFactory.java

```diff
@@ -0,0 +1,114 @@
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * This is an implementation of {@link ScriptEngineFactory} for Jython.
+ *
+ * @author Scott Rushworth - Initial contribution
+ */
+@Component(service = org.openhab.core.automation.module.script.ScriptEngineFactory.class)
+@NonNullByDefault
+public class JythonScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private static final String SCRIPT_TYPE = "py";
+    private static javax.script.ScriptEngineManager ENGINE_MANAGER = new javax.script.ScriptEngineManager();
+    private static final String DEFAULT_PYTHON_PATH = new StringBuilder(ConfigConstants.getConfigFolder())
+            .append(File.separator).append("automation").append(File.separator).append("lib").append(File.separator)
+            .append("python").toString();
+
+    @Activate
+    public JythonScriptEngineFactory() {
+        logger.debug("Loading JythonScriptEngineFactory");
+        String home = JythonScriptEngineFactory.class.getProtectionDomain().getCodeSource().getLocation().toString()
+                .replace("file:", "");
+        System.setProperty("python.home", home);
+
+        String existingPythonPath = System.getProperty("python.path");
+        if (existingPythonPath == null || existingPythonPath.isEmpty()) {
+            System.setProperty("python.path", DEFAULT_PYTHON_PATH);
+        } else if (!existingPythonPath.contains(DEFAULT_PYTHON_PATH)) {
+            TreeSet<String> newPythonPathList = new TreeSet<>(
+                    new ArrayList<String>(Arrays.asList(existingPythonPath.split(File.pathSeparator))));
+            newPythonPathList.add(DEFAULT_PYTHON_PATH);
+            String newPythonPath = String.join(File.pathSeparator, newPythonPathList);
+            System.setProperty("python.path", newPythonPath);
+        }
+
+        StringBuilder newPythonCachedir = new StringBuilder(System.getenv("OPENHAB_USERDATA")).append(File.separator)
+                .append("tmp");
+        System.setProperty("python.cachedir", newPythonCachedir.toString());
+
+        logger.trace("python.home [{}], python.path [{}], python.cachdir [{}]", System.getProperty("python.home"),
+                System.getProperty("python.path"), System.getProperty("python.cachedir"));
+    }
+
+    @Override
+    public List<String> getScriptTypes() {
+        List<String> scriptTypes = new ArrayList<>();
+
+        for (javax.script.ScriptEngineFactory factory : ENGINE_MANAGER.getEngineFactories()) {
+            List<String> extensions = factory.getExtensions();
+
+            if (extensions.contains(SCRIPT_TYPE)) {
+                scriptTypes.addAll(extensions);
+                scriptTypes.addAll(factory.getMimeTypes());
+            }
+        }
+        return Collections.unmodifiableList(scriptTypes);
+    }
+
+    @Override
+    public @Nullable ScriptEngine createScriptEngine(String scriptType) {
```

Comment:
Again - Why do you need this? It is similar to `AbstractScriptEngineFactory#createScriptEngine(String)`.

Reply:
Same reason as above. ENGINE_MANAGER from [ScriptEngineFactory](https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineFactory.java#L35) was not seeing the Jython javax.script.ScriptEngineFactory until after a new instance of javax.script.ScriptEngineManager was created.